docs/developers/local-setup.md

```diff
@@ -82,3 +82,129 @@ If you wish to use a different image URL, you can add the appropriate line to mi
 ### Reconfigure logging
 
 See the information in the "Docker for Desktop" section
+
+## Using kind and local docker registry
+
+### Prerequisites
+
+Complete these prerequisites to run Tekton locally using Kind:
+
+- Install the [required tools](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#requirements).
+- Install [Docker](https://www.docker.com/get-started).
+- Install [kind](https://kind.sigs.k8s.io/).
+
+### Use local registry without authentication
+
+See [Using KinD](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#using-kind).
+
+### Use local private registry
+
+1. Create password file with basic auth.
+
+```bash
+export TEST_USER=testuser
+export TEST_PASS=testpassword
+if [ ! -f auth ]; then
+    mkdir auth
+fi
+docker run \
+ --entrypoint htpasswd \
+ httpd:2 -Bbn $TEST_USER $TEST_PASS > auth/htpasswd
+```
+
+2. Start kind cluster and local private registry
+
+
+Execute the script.
+
+```shell
+#!/bin/sh
+set -o errexit
+
+# create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5000'
+running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+if [ "${running}" != 'true' ]; then
+ docker run \
+   -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
+   -v "$(pwd)"/auth:/auth \
+   -e "REGISTRY_AUTH=htpasswd" \
+   -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+   -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+   registry:2
```

Comment:
Nice :)
I suppose it might be beneficial, at least in some cases, to have the auth setup even for local development.
Perhaps worth replicating in https://github.com/tektoncd/plumbing/blob/ec7bc19a0ce8c2ece2875b714fb69182ee76c776/hack/tekton_in_kind.sh#L65-L73

Reply:
Yeah! I'm not so familiar with plumbing and all our test infra, but I'm trying to learn them as well. 😄